frontend/src/App.tsx

```diff
@@ -18,14 +18,16 @@ export const DataContext = createContext<SingleCasePoint[]>([])
 
 const App: FC = () => {
     const store = useContext(Store);
-    const { surgeryUrgencySelection, outcomeFilter, currentSelectPatientGroup, currentOutputFilterSet } = store.state;
+    const { bloodComponentFilter, surgeryUrgencySelection, outcomeFilter, currentSelectPatientGroup, currentOutputFilterSet, testValueFilter } = store.state;
     const [hemoData, setHemoData] = useState<SingleCasePoint[]>([])
     const [outputFilteredData, setOutputFilteredDAta] = useState<SingleCasePoint[]>([])
 
 
     useEffect(() => {
-        if (process.env.REACT_APP_REQUIRE_LOGIN !== "true") { cacheHemoData(); }
-        if (store.configStore.isLoggedIn && hemoData.length === 0) {
+        if (process.env.REACT_APP_REQUIRE_LOGIN !== "true") {
+            cacheHemoData();
+        }
+        else if (store.configStore.isLoggedIn && hemoData.length === 0) {
```

Comment:
You could simplify this to:

```
if (process.env.REACT_APP_REQUIRE_LOGIN !== "true" || (store.configStore.isLoggedIn && hemoData.length === 0))
```

I'm not sure this is more readable, though. I'll leave it up to you